src/install/lockfile.zig

```diff
@@ -2547,13 +2547,32 @@ pub const Package = extern struct {
                         ),
                     ),
                 );
-                dependency_version.value.workspace = path;
+                defer dependency_version.value.workspace = path;
                 var workspace_entry = try lockfile.workspace_paths.getOrPut(allocator, @truncate(u32, external_name.hash));
                 if (workspace_entry.found_existing) {
-                    log.addErrorFmt(&source, logger.Loc.Empty, allocator, "Workspace name \"{s}\" already exists", .{
-                        external_name.slice(buf),
-                    }) catch {};
-                    return error.InstallFailed;
+                    const old_path = workspace_entry.value_ptr.*;
+
+                    if (strings.eqlComptime(workspace, "*")) {
```

Comment:
Should `*` be special cased? That's my main question here. Or should we just ignore duplicate dependency errors for workspace packages that are also in dependencies.

Reply:
I see `workspace:*` as "whatever you can find under `workspaces: [...]` that fits my bill, use that"

We are essentially ignoring it, as we `return null`, i.e. dropped the dependency instead of creating a new entry under `package.dependencies`